src/main/java/gov/nasa/pds/validate/ri/AuthInformation.java

```diff
@@ -0,0 +1,97 @@
+package gov.nasa.pds.validate.ri;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.Properties;
+import java.util.Scanner;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+public class AuthInformation {
+  final public static AuthInformation NO_AUTH = new AuthInformation(false, "", "" ,"");
+  final private boolean trustSelfSigned;
+  final private String password;
+  final private String url;
+  final private String username;
+
+  private AuthInformation (boolean tss, String pwd, String un, String url) {
+    this.password = pwd;
+    this.trustSelfSigned = tss;
+    this.url = url;
+    this.username = un;
+  }
+
+  public static AuthInformation buildFrom (String filename)
+      throws IOException,ParserConfigurationException,SAXException {
+    boolean tss;
+    File file = new File(filename);
+    Scanner textReader;
+    String line = null, pwd, un, url;
+
+    if (filename == null || filename.length() == 0) return NO_AUTH;
+    if (!file.exists())
+      throw new IOException("Filename '" + filename + "' does not exist");
+
+    // Get the first non-comment line
+    textReader = new Scanner(file,Charset.defaultCharset().name());
+    while (textReader.hasNext() && line == null) {
+      line = textReader.nextLine().strip();
+      if (line.charAt(0) == '#') line = null;
+    }
+    textReader.close();
+    
+    // Determine which file processing to use
+    if (line.startsWith("<?xml ") && line.endsWith("?>")) { // XML
+      // <registry url="http://localhost:9200" index="registry" auth="/path/to/auth.cfg" />
+      DocumentBuilder builder = DocumentBuilderFactory.newDefaultInstance().newDocumentBuilder();
```

Comment:
That lines causes an error in my environment (jdk11):

16:43:56,416  WARN main CommandLineInterface:process:83 - Only the DB will be checked because no registry authorization was given
Exception in thread "main" java.lang.Error: Unresolved compilation problem: 
	The method newDefaultInstance() is undefined for the type DocumentBuilderFactory

	at gov.nasa.pds.validate.ri.AuthInformation.buildFrom(AuthInformation.java:54)
	at gov.nasa.pds.validate.ri.CommandLineInterface.process(CommandLineInterface.java:100)
	at gov.nasa.pds.validate.ReferenceIntegrityMain.main(ReferenceIntegrityMain.java:7)

Then it unclear to me, does this .cfg file passed in arguments of the command line need to be an xml file as for the harvest job configuration ?

That would help if you had some example configuration in src/test/resources/integrity for example.

If we need to have a xml file as an argument then the help message in the command line are misleading, see:
An auth-file is either a text file of the Java property format with two
variables: 'url' and 'credentials'. The 'url' property should be the
complete base URL like 'https://localhost:9876/base' and 'credentials' a
java property file with the user name, password, and other credential
information as that used by harvest. Or it is an XML text file used by
harvest with <registry> containing the 'auth' attribute.